File: backend/app/schemas/audit_event.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AuditEventCreate(BaseModel):
    tenant_id: uuid.UUID
    actor_user_id: uuid.UUID | None = None
    entity_type: str
    entity_id: uuid.UUID
    action: str
    summary: str
    old_values_json: dict | None = None
    new_values_json: dict | None = None
    metadata_json: dict | None = None


class AuditEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    actor_user_id: uuid.UUID | None = None
    entity_type: str
    entity_id: uuid.UUID
    action: str
    summary: str
    old_values_json: dict | None = None
    new_values_json: dict | None = None
    metadata_json: dict | None = None
    created_at: datetime
    updated_at: datetime


class AuditEventListResponse(BaseModel):
    items: list[AuditEventResponse]
    total: int

File: frontend/src/types/auditEvent.ts
export type AuditEventRecord = {
  id: string;
  tenant_id: string;
  actor_user_id: string | null;
  entity_type: string;
  entity_id: string;
  action: string;
  summary: string;
  old_values_json: Record<string, unknown> | null;
  new_values_json: Record<string, unknown> | null;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
};

export type AuditEventListResponse = {
  items: AuditEventRecord[];
  total: number;
};